os/drivers/input/ist415.c

```diff
@@ -1270,14 +1269,26 @@ int ist415_initialize(const char *path, struct i2c_dev_s *i2c, struct ist415_con
 	upper = (struct touchscreen_s *)kmm_zalloc(sizeof(struct touchscreen_s));
 	if (!upper) {
 		ist415dbg("Fail to alloc touchscreen_s\n");
-		kmm_free(dev);
 		ist415_disable(dev);
-		return ERROR;
+		goto cleanup_task_wd;
 	}
 	upper->ops = &g_ist415_ops;
 	upper->priv = dev;
 	dev->upper = upper;
+	ret = touch_register(path, upper);
+	if(ret < 0) {
+		ist415_disable(dev);
+		goto cleanup_task_wd;
```

Comment:
```suggestion
		kmm_free(upper);
		goto cleanup_task_wd;
```
we need to release upper.

Reply:
Could you please add fail log here?

Reply:
fixed

Reply:
Could you please add fail log here?

Reply:
> Could you please add fail log here?

Its already there in touchscreen.c function
should I add it here also?

Reply:
okay, touch_regiseter is printing fail case log,
we don't need to add log here!